src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        int[] arrayResultInt = new int[names.length];
+        List<String> list = new ArrayList<>(List.of(names));
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate endDate = LocalDate.parse(dateTo, formatter);
+        for (String s : data) {
+            String[] split = s.split(" ");
+            LocalDate localDate = LocalDate.parse(split[0], formatter);
+            if ((localDate.isAfter(startDate) || localDate.isEqual(startDate))
+                    && (localDate.isBefore(endDate) || localDate.isEqual(endDate))) {
+                int salarySum = Integer.parseInt(split[2]) * Integer.parseInt(split[3]);
+                arrayResultInt[list.indexOf(split[1])] += salarySum;
```

Comment:
All magic numbers(0, 1, 2, 3), strings, symbols(" ", " - ") in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)